### ISSUE TITLE
Can't parse `foo! += 1`

### ISSUE THREAD

**Issue by paoloricciuti** [TyrealHu/acorn-typescript#64]
If i try to parse the expression in the title i get an error despite it being a [valid typescript syntax](https://www.typescriptlang.org/play/?#code/DYUwLgBAZg9jBcEB2BXAtgIxAJwgH2RWGAgF4IAGAbgCgbYYBCCAanIEYqg).

Syntax:
```ts
let foo: number | null  = 1;
foo! += 1;
```
Error:
```
Error: Assigning to rvalue (3:0)
```
Repo:
[link](https://stackblitz.com/edit/node-jgz7dz?file=index.js)

Related issue:

https://github.com/sveltejs/svelte/issues/13409

We are using `acorn-typescript` in `svelte` to parse typescript scripts and this bug was reported.

**Comment by dummdidumm** [comment]
Fixed in https://github.com/sveltejs/acorn-typescript
